frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/GitCommandUpdatingCurrentBranchBackgroundable.java

```diff
@@ -0,0 +1,206 @@
+      if (updatedRanges != null &&
+          AbstractCommonUpdateAction
+              .showsCustomNotification(java.util.Collections.singletonList(GitVcs.getInstance(project)))) {
+        var ranges = updatedRanges.calcCurrentPositions();
+        GitUpdateInfoAsLog.NotificationData notificationData = new GitUpdateInfoAsLog(project, ranges)
+            .calculateDataAndCreateLogTab();
+
+        Notification notification;
+        if (notificationData != null) {
+          String title = getTitleForUpdateNotification(notificationData.getUpdatedFilesCount(),
+              notificationData.getReceivedCommitsCount());
+          String content = getBodyForUpdateNotification(notificationData.getFilteredCommitsCount());
+          notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(title,
+              content,
+              INFORMATION,
+              /* listener */ null);
+          notification.addAction(NotificationAction.createSimple(getString(
+              "action.GitMachete.GitCommandUpdatingCurrentBranchBackgroundable.notification.message.view-commits"),
+              notificationData.getViewCommitAction()));
+
+        } else {
+          // When the pull results with no commits, there is no git update info (as log).
+          // Based on that we know that all files are up-to-date.
+          notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
+              getString(
+                  "action.GitMachete.GitCommandUpdatingCurrentBranchBackgroundable.notification.title.all-files-are-up-to-date"),
+              /* content */ "", INFORMATION, /* listener */ null);
+        }
+        VcsNotifier.getInstance(project).notify(notification);
+
+      } else {
+        showUpdates(root, currentRev, beforeLabel);
+      }
+
+    } else if (localChangesDetector.wasMessageDetected()) {
+      LocalChangesWouldBeOverwrittenHelper.showErrorNotification(project,
+          gitRepository.getRoot(),
+          getOperationName(),
+          localChangesDetector.getRelativeFilePaths());
+
+    } else if (untrackedFilesDetector.wasMessageDetected()) {
+      GitUntrackedFilesHelper.notifyUntrackedFilesOverwrittenBy(project,
+          root,
+          untrackedFilesDetector.getRelativeFilePaths(),
+          getOperationName(),
+          /* description */ null);
+
+    } else {
+      GitUIUtil.notifyError(project,
+          format(getString("action.GitMachete.GitCommandUpdatingCurrentBranchBackgroundable.notification.fail"),
+              getOperationName()),
+          result.getErrorOutputAsJoinedString(),
+          /* important */ true,
+          /* error */ null);
+      gitRepository.update();
+    }
+  }
+
+  private void showUpdates(VirtualFile root, GitRevisionNumber currentRev, Label beforeLabel) {
+    try {
+      UpdatedFiles files = UpdatedFiles.create();
+      @SuppressWarnings("UnstableApiUsage") var collector = new MergeChangeCollector(project, root, currentRev);
```

Comment:
`MergeChangeCollector`:
```
@Deprecated
@ApiStatus.ScheduledForRemoval(inVersion = "2020.3")
```
:confused:

Reply:
Whoops @mkondratek do we have a remedy for this one? :confused:

Reply:
We can simply replace the `VirtualFile root` with `GitRepository repository` (the suggested constructor)

![image](https://user-images.githubusercontent.com/19799111/90425891-c9ab2280-e0c0-11ea-982f-e0afff806b77.png)

Reply:
Whoops asked before checking first, sorry :xd:

Reply:
Done, fixed

Reply:
Whoops not so easy, the newer c'tor isn't available on 2020.1 yet... let me do some reflection magic to provide smooth transition to 2020.3